### PR TITLE
unhide disable-downsampling flag in compactor

### LINKED ISSUE #1377: Unwanted downsampling during Thanos compact
**Thanos, Prometheus and Golang version used**

thanos, version 0.6.0 (branch: HEAD, revision: c70b80eb83e52f5013ed5ffab72464989c132883)
  build user:       circleci@090532e54e70
  build date:       20190718-11:16:14
  go version:       go1.12.5

**What happened**

I just wanted to compact the data in the s3 bucket and delete only data (3d just for test environment) 
 and found that also down sampling occurs!
I don't want down sampled data.

I used the following command:
```
/ $ thanos compact \
>  --data-dir=/tmp \
>  --objstore.config-file=/creds/bucket_config \
>  --http-address="0.0.0.0:10900" \
>  --log.level=debug \
>  --retention.resolution-raw=3d 
```
e.g. output:
<details>Logs
<p>
```
level=info ts=2019-08-06T09:54:03.850088824Z caller=main.go:154 msg="Tracing will be disabled"
level=info ts=2019-08-06T09:54:03.850702342Z caller=factory.go:39 msg="loading bucket configuration"
level=info ts=2019-08-06T09:54:03.878902477Z caller=compact.go:231 msg="retention policy of raw samples is enabled" duration=24h0m0s
level=info ts=2019-08-06T09:54:03.879200381Z caller=compact.go:325 msg="starting compact node"
level=info ts=2019-08-06T09:54:03.879325966Z caller=main.go:326 msg="Listening for metrics" address=0.0.0.0:10902
level=info ts=2019-08-06T09:54:03.879393724Z caller=compact.go:1023 msg="start sync of metas"
level=debug ts=2019-08-06T09:54:03.9523658Z caller=compact.go:252 msg="download meta" block=01DHJVMHJEXVST2FR6EW5PNM9X
level=debug ts=2019-08-06T09:54:03.952394471Z caller=compact.go:252 msg="download meta" block=01DHK53ATT2ZNNPJHCHA3569AR
level=debug ts=2019-08-06T09:54:03.952406047Z caller=compact.go:252 msg="download meta" block=01DHJMRTADXZ3RHF5YSNQQ6KWE
level=debug ts=2019-08-06T09:54:03.952650382Z caller=compact.go:252 msg="download meta" block=01DHK2G8TE1N132SY99EY83RNZ
level=debug ts=2019-08-06T09:54:03.952672481Z caller=compact.go:252 msg="download meta" block=01DHJDX32DF5GYSGAM8H0XQ8CJ
level=info ts=2019-08-06T09:54:04.147372456Z caller=compact.go:1029 msg="start of GC"
level=info ts=2019-08-06T09:54:04.147474988Z caller=compact.go:1035 msg="start of compaction"
level=info ts=2019-08-06T09:54:04.153305811Z caller=compact.go:244 msg="compaction iterations done"
level=info ts=2019-08-06T09:54:04.153334855Z caller=compact.go:251 msg="start first pass of downsampling"
level=info ts=2019-08-06T09:54:06.08809803Z caller=downsample.go:258 msg="downloaded block" id=01DHK53ATT2ZNNPJHCHA3569AR duration=1.637697616s
level=info ts=2019-08-06T09:54:08.084805313Z caller=streamed_block_writer.go:219 msg="finalized downsampled block" mint=1564905600000 maxt=1565049600000 ulid=01DHK5K8MR2VJ3SD07906CEPHM resolution=300000
level=info ts=2019-08-06T09:54:08.084886081Z caller=downsample.go:285 msg="downsampled block" from=01DHK53ATT2ZNNPJHCHA3569AR to=01DHK5K8MR2VJ3SD07906CEPHM duration=1.984543979s
level=info ts=2019-08-06T09:54:10.252806399Z caller=downsample.go:299 msg="uploaded block" id=01DHK5K8MR2VJ3SD07906CEPHM duration=2.160810818s
level=info ts=2019-08-06T09:54:10.260123184Z caller=compact.go:257 msg="start second pass of downsampling"
level=info ts=2019-08-06T09:54:10.35408216Z caller=compact.go:262 msg="downsampling iterations done"
level=info ts=2019-08-06T09:54:10.35411316Z caller=retention.go:17 msg="start optional retention"
level=info ts=2019-08-06T09:54:10.549622592Z caller=retention.go:46 msg="optional retention apply done"
level=info ts=2019-08-06T09:54:10.549741209Z caller=main.go:202 msg=exiting
```
</p>
</details>
(Note: at the run all data was younger than 3 days)

If I add retention for the resolutions of the down sampled data e.g. via
```
>  --retention.resolution-5m=1m \
>  --retention.resolution-1h=1m
```
the down sampling still occurs but all down sampled data is immediately deleted at the end.
So it this case all raw data would be down sampled during all compact runs.

<details>Logs
<p>
```
level=info ts=2019-08-06T14:12:07.057353975Z caller=main.go:154 msg="Tracing will be disabled"
level=info ts=2019-08-06T14:12:07.057547083Z caller=factory.go:39 msg="loading bucket configuration"
level=info ts=2019-08-06T14:12:07.947140214Z caller=compact.go:231 msg="retention policy of raw samples is enabled" duration=48h0m0s
level=info ts=2019-08-06T14:12:07.947197423Z caller=compact.go:234 msg="retention policy of 5 min aggregated samples is enabled" duration=1m0s
level=info ts=2019-08-06T14:12:07.947219395Z caller=compact.go:237 msg="retention policy of 1 hour aggregated samples is enabled" duration=1m0s
level=info ts=2019-08-06T14:12:07.947602493Z caller=compact.go:325 msg="starting compact node"
level=info ts=2019-08-06T14:12:07.947860518Z caller=compact.go:1023 msg="start sync of metas"
level=info ts=2019-08-06T14:12:07.948759455Z caller=main.go:326 msg="Listening for metrics" address=0.0.0.0:10902
level=debug ts=2019-08-06T14:12:08.647702018Z caller=compact.go:252 msg="download meta" block=01DHKCFRYPP85BXE712P0ZKD4W
level=debug ts=2019-08-06T14:12:08.647750969Z caller=compact.go:252 msg="download meta" block=01DHKG7QADSAKRGC1YESTXTR9C
level=debug ts=2019-08-06T14:12:08.647726937Z caller=compact.go:252 msg="download meta" block=01DHK53ATT2ZNNPJHCHA3569AR
level=debug ts=2019-08-06T14:12:08.648319554Z caller=compact.go:252 msg="download meta" block=01DHK9C02E59EPFZB025ZWYJKJ
level=info ts=2019-08-06T14:12:08.949005831Z caller=compact.go:1029 msg="start of GC"
level=info ts=2019-08-06T14:12:08.949076082Z caller=compact.go:1035 msg="start of compaction"
level=info ts=2019-08-06T14:12:08.954315723Z caller=compact.go:244 msg="compaction iterations done"
level=info ts=2019-08-06T14:12:08.954343061Z caller=compact.go:251 msg="start first pass of downsampling"
level=info ts=2019-08-06T14:12:10.752781985Z caller=downsample.go:258 msg="downloaded block" id=01DHK53ATT2ZNNPJHCHA3569AR duration=1.700999478s
2019/08/06 14:12:18 Unsolicited response received on idle HTTP channel starting with "HTTP/1.0 408 Request Time-out\r\nCache-Control: no-cache\r\nConnection: close\r\nContent-Type: text/html\r\n\r\n<html><body><h1>408 Request Time-out</h1>\nYour browser didn't send a complete request in time.\n</body></html>\n"; err=<nil>
level=info ts=2019-08-06T14:12:36.545989261Z caller=streamed_block_writer.go:219 msg="finalized downsampled block" mint=1564905600000 maxt=1565049600000 ulid=01DHKMBTVNEYWQD7NBD5G93PTA resolution=300000
level=info ts=2019-08-06T14:12:36.546082588Z caller=downsample.go:285 msg="downsampled block" from=01DHK53ATT2ZNNPJHCHA3569AR to=01DHKMBTVNEYWQD7NBD5G93PTA duration=25.298216855s
level=info ts=2019-08-06T14:12:38.452506241Z caller=downsample.go:299 msg="uploaded block" id=01DHKMBTVNEYWQD7NBD5G93PTA duration=1.702983802s
level=info ts=2019-08-06T14:12:38.548204941Z caller=compact.go:257 msg="start second pass of downsampling"
level=info ts=2019-08-06T14:12:38.646558266Z caller=compact.go:262 msg="downsampling iterations done"
level=info ts=2019-08-06T14:12:38.646617081Z caller=retention.go:17 msg="start optional retention"
level=info ts=2019-08-06T14:12:38.75118026Z caller=retention.go:35 msg="deleting block" id=01DHKMBTVNEYWQD7NBD5G93PTA maxTime="2019-08-06 00:00:00 +0000 UTC"
level=info ts=2019-08-06T14:12:39.247645798Z caller=retention.go:46 msg="optional retention apply done"
level=info ts=2019-08-06T14:12:39.247751952Z caller=main.go:202 msg=exiting
```
</p>
</details>

**What you expected to happen**

I didn't expect downsampling to occur by command "compact".
I thought for this the command "downsample" is provided separately.

How I can I compact the data without downsampling?

**How to reproduce it (as minimally and precisely as possible)**:

see above 

**Full logs to relevant components**

see above.

**Anything else we need to know**

-

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

* [] CHANGELOG entry if change is relevant to the end user.

fix #1377

Do I have to remove the `debug` prefix of this flag?

### CODE REVIEW

File: cmd/thanos/compact.go [hunk 1]
```diff
@@ -100,9 +100,9 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application) {
 		Hidden().Default("false").Bool()
 
 	// TODO(bplotka): Remove this flag once https://github.com/thanos-io/thanos/issues/297 is fixed.
```
Comment (GiedriusS) [inline cmd/thanos/compact.go]: Does it still make sense for us to have this comment?

Comment (yeya24) [reply cmd/thanos/compact.go]: I think it is better to remove this comment.

File: cmd/thanos/compact.go [hunk 2]
```diff
@@ -99,10 +99,9 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application) {
 	generateMissingIndexCacheFiles := cmd.Flag("index.generate-missing-cache-file", "If enabled, on startup compactor runs an on-off job that scans all the blocks to find all blocks with missing index cache file. It generates those if needed and upload.").
 		Hidden().Default("false").Bool()
 
-	// TODO(bplotka): Remove this flag once https://github.com/thanos-io/thanos/issues/297 is fixed.
-	disableDownsampling := cmd.Flag("debug.disable-downsampling", "Disables downsampling. This is not recommended "+
+	disableDownsampling := cmd.Flag("disable-downsampling", "Disables downsampling. This is not recommended "+
```
Comment (bwplotka) [inline cmd/thanos/compact.go]: Can we name it `downsampling.disable`?

File: cmd/thanos/compact.go [hunk 3]
```diff
@@ -99,10 +99,9 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application) {
...
 		"as querying long time ranges without non-downsampled data is not efficient and not useful (is not possible to render all for human eye).").
```
Comment (bwplotka) [inline cmd/thanos/compact.go]:
```suggestion
		"as querying long time ranges without non-downsampled data is not efficient and useful e.g it is not possible to render all samples for a human eye anyway").
```

File: cmd/thanos/compact.go (hunk 2 above)
Comment (bwplotka) [reply cmd/thanos/compact.go]: Essentially we might want to have more `downsampling` options some day (:

**Comment by yeya24** [comment]
Updated :smile: PTAL @bwplotka  @GiedriusS

**Comment by bwplotka** [comment]
Thank you! LGTM.

**Pull request merged**